packages/darklang/languageTools/lsp-server/docSync.dark

```diff
@@ -31,31 +31,46 @@ module Darklang =
                     ) }
             )
 
+        let parseAndreportDiagnostics
+          (requestText: String)
+          : Stdlib.Result.Result<WrittenTypes.ParsedFile, String> =
+          let onMissing = LanguageTools.NameResolver.OnMissing.Allow
+          let pm = LanguageTools.PackageManager.pm ()
+
+          let parsed =
+            requestText |> Parser.parseToSimplifiedTree |> Parser.parseFromTree
+
+          let sourceFile =
+            match parsed with
+            | Ok parsedFile ->
+              match parsedFile with
+              | SourceFile sf -> sf
+              | _ -> log "Invalid source file"
+            | Error err -> log err
+
+          let (_ptCliScriptModule, unresolvedNames) =
+            Parser.CliScript.parse
```

Comment:
Is it ok to reuse `CliScript.parse` here?

Reply:
Yes - I think that's probably ideal, for now - just focusing on .dark "scripts." Once the LS has a virtual document system set up, where we can be "editing the contents of X package" or something else, we can adjust here accordingly.